packages/nitro-protocol/contracts/examples/RockPaperScissors.sol

```diff
@@ -87,75 +98,146 @@ contract RockPaperScissors is ForceMoveApp {
-        }
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    ) private pure {
+        // TODO
+    }
 
     function requireValidACCEPT(
-        VariablePart memory a,
-        VariablePart memory b,
-        RPSData memory appDataA,
-        RPSData memory appDataB
-        ) private pure
-        {
-            // TODO
-        }
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    ) private pure {
+        // TODO
+    }
+
+    function requireValidREVEAL(
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    ) private pure {
+        uint256 playerFirstWinnings;
+        uint256 playerSecondWinnings;
+        require(
+            toGameData.playerSecondWeapon == fromGameData.playerSecondWeapon,
+            "Player Second's weapon should be the same between commitments."
+        );
+
+        // check hash matches
+        // need to convert Weapon -> uint256 to get hash to work
+        bytes32 hashed = keccak256(
+            abi.encodePacked(uint256(toGameData.playerFirstWeapon), toGameData.salt)
+        );
+        require(hashed == fromGameData.preCommit, 'The hash needs to match the precommit');
+
+        // calculate winnings
+        (playerFirstWinnings, playerSecondWinnings) = winnings(
+            toGameData.playerFirstWeapon,
+            toGameData.playerSecondWeapon,
+            toGameData.stake
+        );
+
+        Outcome.OutcomeItem[] memory outcomeFrom = abi.decode(fromPart.outcome, (Outcome.OutcomeItem[]));
+        Outcome.OutcomeItem[] memory outcomeTo = abi.decode(toPart.outcome, (Outcome.OutcomeItem[]));
+        Outcome.AssetOutcome memory assetOutcomeFrom = abi.decode(
+            outcomeFrom[0].assetOutcomeBytes,
+            (Outcome.AssetOutcome)
+        );
+        Outcome.AssetOutcome memory assetOutcomeTo = abi.decode(
+            outcomeTo[0].assetOutcomeBytes,
+            (Outcome.AssetOutcome)
+        );
+        Outcome.AllocationItem[] memory allocationFrom = abi.decode(
+            assetOutcomeFrom.allocationOrGuaranteeBytes,
+            (Outcome.AllocationItem[])
+        );
+        Outcome.AllocationItem[] memory allocationTo = abi.decode(
+            assetOutcomeTo.allocationOrGuaranteeBytes,
+            (Outcome.AllocationItem[])
+        );
+
+        require(
+            allocationTo[0].amount == allocationFrom[0].amount.add(playerFirstWinnings),
+            "Player First's allocation should be updated with the winnings."
+        );
+        require(
+            allocationTo[1].amount ==
+                allocationFrom[1].amount.sub(fromGameData.stake.mul(2)).add(playerSecondWinnings),
+            "Player Second's allocation should be updated with the winnings."
+        );
```

Comment:
We need to require that the the destinations are not changing, *globally* across all transitions.

i.e. `allocationTo[0].destination == allocationFrom[0].destination`